lua/lib/reflection.lua

```diff
@@ -212,12 +228,13 @@ function reflection:begin_playback()
           if self.loop == 1 then
             self.step = 0
             self:_clear_flags()
-            self:start_callback()
+            queued_start_callback = true
           end
         end
       end
     else
-      if self.step % q ~= 1 then goto continue end
+      if q ~= 1 and self.step % q ~= 1 then goto continue end
+      self.step_callback()
       for i = q - 1, 0, -1 do
         if self.event[self.step - i] and next(self.event[self.step - i]) then
           for j = 1, #self.event[self.step - i] do
```

Comment:
This line and the next could be made shorter, I think, by instead just saying
```lua
for _, event in ipairs(self.event[self.step - i]) do
  if not event._flag then self.process(event) end
end
```

Reply:
yesssss